src/Raven.Server/Documents/DatabasesLandlord.cs

```diff
@@ -171,14 +171,19 @@ private void NotifyPendingClusterTransaction(string databaseName, Task<DocumentD
                 }
                 return;
             }
+
+            RavenTransaction rtnCtx = null;
             try
             {
                 using (_serverStore.ContextPool.AllocateOperationContext(out TransactionOperationContext serverContext))
                 using (database.DocumentsStorage.ContextPool.AllocateOperationContext(out DocumentsOperationContext docContext))
-                using (serverContext.OpenReadTransaction())
                 using (docContext.OpenReadTransaction())
                 {
-                    ExecutePendingClusterTransactions(database, index, docContext, serverContext);
+                    rtnCtx = serverContext.OpenReadTransaction();
+                    var tasks = ExecutePendingClusterTransactions(database, index, docContext, serverContext);
```

Comment:
This looks like it is wrong to me. You are _not_ allowed to use a single transaction concurrently from multiple threads.
But you are spawning off multiple tasks here?